src/Microsoft.ApiDesignGuidelines.Analyzers/CSharp/CSharpUseIntegralOrStringArgumentForIndexers.cs

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Diagnostics;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.ApiDesignGuidelines.Analyzers
+{
+    /// <summary>
+    /// CA1043: UseIntegralOrStringArgumentForIndexers
+    /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class CSharpUseIntegralOrStringArgumentForIndexersAnalyzer : UseIntegralOrStringArgumentForIndexersAnalyzer
```

Comment:
You don't need this file since your analyzer is a symbol analyzer. Make the base type sealed and delete the C#\VB files.